ray-operator/controllers/ray/common/pod.go

```diff
@@ -753,10 +759,17 @@ func generateRayStartCommand(ctx context.Context, nodeType rayv1.RayNodeType, ra
 	if _, ok := rayStartParams["num-gpus"]; !ok {
 		// Scan for resource keys ending with "gpu" like "nvidia.com/gpu".
 		for resourceKey, resource := range resource.Limits {
-			if strings.HasSuffix(string(resourceKey), "gpu") && !resource.IsZero() {
+			resourceKeyString := string(resourceKey)
+			if strings.HasSuffix(resourceKeyString, "gpu") && !resource.IsZero() {
 				rayStartParams["num-gpus"] = strconv.FormatInt(resource.Value(), 10)
 				// For now, only support one GPU type. Break on first match.
```

Comment:
This comment and `break` below is now no longer true. I'm not sure how common it is to have nodes that have both GPUs and custom accelerators, but I think the `break` can be removed

Reply:
The comment can be removed, but I think `break` is needed to match the first `resourceKey` and then exit of the loop, instead of parsing both and passing them to the `ray start`. This is assuming that either `num-gpus` can exist or the custom accelerator types.

https://docs.ray.io/en/latest/ray-core/scheduling/accelerators.html

Reply:
Is there somewhere in the docs that says you can't specify `--num-gpus` and a custom accelerator?

I think my main point here is that if a user specifies `--resources` in rayStartParams today with AWS neuron cores, it should override the new behavior of parsing the container resource limits. Right now the behavior only applies if `--num-gpus` is not specified which I think is incorrect.

Reply:
Consider changing the code to something like this:
```go
if _, ok := rayStartParams["num-gpus"]; !ok {
    // Scan for resource keys ending with "gpu" like "nvidia.com/gpu".
    for resourceKey, resource := range resource.Limits {
	    resourceKeyString := string(resourceKey)
	    if strings.HasSuffix(resourceKeyString, "gpu") && !resource.IsZero() {
		    rayStartParams["num-gpus"] = strconv.FormatInt(resource.Value(), 10)
		    // For now, only support one GPU type. Break on first match.
		    break
	    }
    }
}

if _, ok := rayStartParams["resources"]; !ok {
  for resourceKey, resource := range resource.Limits {
      resourceKeyString := string(resourceKey)
      if rayResourceName, ok := customAcceleratorToRayResourceMap[resourceKeyString]; ok && !resource.IsZero() {
            if err := addCustomAcceleratorToResourcesIfNotExists(rayStartParams, rayResourceName, resource.Value()); err != nil {
                log.Error(err, fmt.Sprintf("failed to add %s to resources", rayResourceName))
            }
            // For now, only support one custom accelerator type. Break on first match.
            break
      }
  }
}
```

Reply:
The documentation does not say that it supports both `gpu` and a custom accelerator in combination either. So I am not sure if we should allow parsing both. Wdyt?

And for the code change you suggested above, we do not want to override resources if its specified already? Is that right? Can other params be passed as part of resources or just custom accelerator types, and we do not want to add eg: `neuron_cores` based on the resourceLimits specified?

Reply:
> And for the code change you suggested above, we do not want to override resources if its specified already? Is that right? Can other params be passed as part of resources or just custom accelerator types, and we do not want to add eg: neuron_cores based on the resourceLimits specified?

I suggest following the same pattern as `num-gpus` for custom resources. We should only set it if not specified in rayStartParam but specified in as a container resource limit. If specified in start params, we shouldn't override it.

Reply:
> The documentation does not say that it supports both gpu and a custom accelerator in combination either.

Technically, Ray wouldn't have an issue with that.
But it would be unusual for a cloud provider to support attaching a provider-specific chip and an nvidia gpu to a single VM.